localstack/utils/kinesis/kinesis_connector.py

```diff
@@ -305,7 +306,7 @@ def get_stream_info(
     if aws_stack.is_local_env(env):
         stream_info["conn_kwargs"] = {
             "host": LOCALHOST,
-            "port": config.service_port("kinesis"),
+            "port": localstack_host().port,
```

Comment:
Since this happens locally, shouldn't we use the gateway listen port here?

Reply:
Yeah that explains the hard coding of `LOCALHOST` as well :+1:

Reply:
There are lots of hardcoded `LOCALHOST` and `LOCALHOST_HOSTNAME` examples for returning URLs that might need a follow-up as well 😬 ...

Reply:
Do you have any examples? I don't immediately see anything that I missed

Reply:
For example this one mixing `localhost.localstack.cloud` (external) with `EDGE_PORT` (internal):
https://github.com/localstack/localstack/blob/ccb95521b867a35576e9ad3d61e620fbd761f015/localstack/utils/testutil.py#L689

The API gateway tests have similar occurrences, for example here:
https://github.com/localstack/localstack/blob/ccb95521b867a35576e9ad3d61e620fbd761f015/tests/aws/services/apigateway/test_apigateway_basic.py#L1223


I rebased the edge port migration PRs upon these PRs fixing these issues and other `EDGE_PORT` occurrences I spotted while reviewing this PR. We should not delay this any further.

Reply:
https://github.com/localstack/localstack/pull/9633